### PR TITLE
Create household cars from Vehicle types

### PR DESCRIPTION
**Pull request by haitamlaarabi**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3456)

### CODE REVIEW

**Comment by haitamlaarabi** [comment]
Test!

File: src/main/scala/beam/sim/config/BeamConfig.scala
Comment (JustinPihony) [inline src/main/scala/beam/sim/config/BeamConfig.scala:1531]: You are missing the template.conf change that caused this.

File: src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala
Comment (JustinPihony) [inline src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala:610]: This can be removed I assume?

File: src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala
Comment (JustinPihony) [inline src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala:682]: Per the name, this test should also be testing against the vehicle, shouldn't it?

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala]: Can we set the variable name to `person`? I'm never a fan of shortening if it hurts readability

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala:216]: I've seen similar code before - should we create a central place that lists "AllCategories" for if/when it changes. Otherwise we'll have to do a copy/replace

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala]: Same naming comment here. vs should be vehicles. I know that code was existing, but I don't see why we can't fix it up :)

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala]: Should we log a warning here with the type attempted?

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala:583]: Does this not have a first pass attempt the way a car does?

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala]: Should we log the vehicle details? Maybe even the state of `vehiclesInternal`?

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala]: The warning logger should be moved up to this level and then you don't need the comment.

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala:108]: Should we create the vehicle here? This should probably be pushed into the if, because if there are available vehicles then we hit the creation code for nothing.

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala:148]: You can change the if to be a match and then the Nil case moves from an unnecessary logging to being the primary emergency generation. Note that you could still end up with a None vehicle in the scenario where no available vehicles and no emergency generator is configured to be available. How you want to handle that (logging?) is up to you.

**Review by JustinPihony — CHANGES_REQUESTED** [review]
There are some items flagged which need handled before approval

**Comment by haitamlaarabi** [comment]
Test now!

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala:583]: Not required

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala:108]: Done

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala:148]: It's fine to end up with a None vehicle

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala:216]: There is already a central place with all categories. This is a special case.

File: src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala
Comment (haitamlaarabi) [reply src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala:682]: Not sure how to answer this

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala]: Done

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala]: okay

**Comment by haitamlaarabi** [comment]
Test now!

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (nikolayilyin) [inline src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala]: a duplication of debug log entries

**Review by nikolayilyin — APPROVED** [review]
a nitpick, otherwise lgtm

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by JustinPihony** [comment]
> You added this into the `if` as per my last comment, but didn't remove it from the `for`

File: src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala
Comment (JustinPihony) [reply src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala:216]: I guess what I meant is that instead of putting `List(Car, Bike)` directly to store it as a variable for use anywhere else it is needed:

`val allVehicleCategories = List(Car, Bike)`

File: src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala:154]: I am just now noticing that this is an `Option[Unit]`, which means it is basically working off of the fact whether the `for` did something at all. Is that what is wanted? If so, then maybe it is better to assign to a variable and then do an if isEmpty. Ultimately it is more about confusing a reader of the code, so it will be functionally the same and not worth blocking., though

**Review by JustinPihony — APPROVED** [review]
Approving, as my changes are not worth blocking, but maybe should be addressed at least

File: src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala
Comment (JustinPihony) [reply src/test/scala/beam/agentsim/agents/PersonWithPersonalVehiclePlanSpec.scala:682]: I guess I was wondering if there is a way to create an assertion that a last resort car was created? A stub that could be verified that the emergency creation was triggered - or a test of the vehicle attributes?

**Pull request merged**
